addon/model.js

```diff
@@ -469,6 +469,10 @@ export default class MegamorphicModel extends EmberObject {
   }
 
   get id() {
+    if (!this._init) {
+      return null;
+    }
+
```

Comment:
We don't want to do this.  It isn't really valid to ask for the `id` of an uninitialized model (or to ask for anything else for that matter).

The reason we're seeing the issue in the inspector is because it calls `toString` on the prototype to get a description of the class name (instead of `constructor.name or similar).

It'll be a lot clearer to handle this in `toString` and not `id`.  Since the result is being used as a sub-header in the properties pane, returning the class name is probably the most useful.

```js
  toString() {
    if(this === this.constructor.prototype) {
      return 'MegamorphicModel';
    }

    return `<MegamorphicModel:${this.id}>`;
  }
```

As far as testing goes i think it's fine for a unit test POV to confirm that `MegamorphicModel.prototype.toString() === 'MegamorphicModel'`

Reply:
Hmm...that makes a lot of sense, also it will render accurately in the Inspector, I believe. I will refactor my approach then. Thanks!